core/src/main/java/org/elasticsearch/search/slice/SliceBuilder.java

```diff
@@ -194,9 +196,14 @@ public Query toFilter(QueryShardContext context, int shardId, int numShards) {
             throw new IllegalArgumentException("field " + field + " not found");
         }
 
+        String field = this.field;
         boolean useTermQuery = false;
         if (UidFieldMapper.NAME.equals(field)) {
-           useTermQuery = true;
+            if (context.getIndexSettings().isSingleType()) {
```

Comment:
as a followup I wonder if we can use the seq ID for this now since it has docvalues? That would be a much better default IMO, at least for 6.0.0 indices